### PR TITLE
Modify tests for more deterministic behavior

### PR DESCRIPTION
**Pull request by michaelvanstraten**

### CODE REVIEW

**Comment by Luni-4** [comment]
@michaelvanstraten 

Now that #1078 has been merged, we can fix CI using this PR. Let's use `yaml` comparisons and not `json` because of the reccomandation written [here](https://docs.rs/insta/1.34.0/insta/macro.assert_json_snapshot.html).

We need to update the snapshots accordingly.

**Comment by michaelvanstraten** [comment]
> Let's use `yaml` comparisons

Already implemented, for the same reason.

> We can also integrate the missing features present in #1080

Do you mean the switch to `std::collections::Hashmap`, the sorting of maps or both?

**Comment by Luni-4** [comment]
We can just sort out the maps in this PR, thanks for the catch!

**Comment by michaelvanstraten** [comment]
@Luni-4, tests pass except for the three sitter checks, can you check it out?

**Review by Luni-4 — CHANGES_REQUESTED** [review]
Thanks a lot! For what concerns `tree-sitter` checks, it seems it's running them because of some changed values. We can leave them as-is for now

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "tests/repositories/rca-output"]
 	path = tests/repositories/rca-output
-	url = https://github.com/mozilla/rust-code-analysis-output
+	url = https://github.com/michaelvanstraten/rust-code-analysis-output
```
Comment (Luni-4) [inline .gitmodules:3]: We have to revert this change

**Review by Luni-4 — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 1 above)
Comment (michaelvanstraten) [reply .gitmodules:3]: https://github.com/mozilla/rust-code-analysis-output/pull/4

Comment (Luni-4) [reply .gitmodules:3]: Merged, thanks! Let's update this PR

**Review by Luni-4 — APPROVED** [review]
Thank you @michaelvanstraten! 

@marco-c In my opinion we can merge this PR and open an issue to find the possible issues with grammars
